delta-app/src/main/java/io/cdap/delta/app/QueueingEventEmitter.java

```diff
@@ -64,7 +64,8 @@ public void emit(DDLEvent event) {
     }
 
     try {
-      eventQueue.put(new Sequenced<>(event, ++sequenceNumber));
+      // don't assign sequence number to DDL event
+      eventQueue.put(new Sequenced<>(event, -1));
```

Comment:
Assigning `-1` here seems error prone. It still means that target can access sequence number for DDL events using API `Sequence.getSequenceNumber()` and accidentally use it. We need to make sure that target is aware of the contract `DDL Events wont have sequence number`. One possible way may be `Sequenced` class can be modified as :
```java
public class Sequenced {
   private final int UNSUPPORTED_SEQUENCE = -1;
  public Sequenced(T event) {
    this(event, UNSUPPORTED_SEQUENCE);
  } 
   public Sequenced(T event, long sequenceNumber) {
    this.event = event;
    this.sequenceNumber = sequenceNumber;
  }
public long getSequenceNumber() {
    if (sequenceNumber == UNSUPPORTED_SEQUENCE) {
        throw new UnsupportedOperationExcetpion();
    }
    return sequenceNumber;
  }
}
```

then in here we can do `evenrQueue.put(new Sequenced<>(event))`

Reply:
good suggestion :). Done :)